src/types/zodFlaw.ts

```diff
@@ -250,4 +250,13 @@ export const ZodFlawSchema = z.object({
       ['affects', `${affect.index}`, 'ps_component'],
     );
   }
+
+  const unregisteredComponents = zodFlaw.components.filter(
+    (component) => !zodFlaw.affects.find(affect => affect.ps_component === component)
+  );
+
+  if (unregisteredComponents.length) {
+    const verb = unregisteredComponents.length === 1 ? 'is' : 'are';
+    raiseIssue(`${zodFlaw.components.join(', ')} ${verb} not in the Flaw's Affected Offerings`, ['components']);
+  }
```

Comment:
When creating the flaw, affects does not exist yet. raising the issue preventing the flaw from creation.

This could be fixed by checking if affects is empty, but I'm not sure if it is what we want. Maybe the check should be to only allow affect with components listed in the flaw? Instead of only allow flaw components that are in the affects,